src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -926,8 +925,14 @@ private FunctionExpression GetLoopItemVariableExpression(ForSyntax @for, Languag
 
         private LanguageExpression ConvertVariableAccess(VariableAccessSyntax variableAccessSyntax)
         {
-            string name = variableAccessSyntax.Name.IdentifierName;
+            var name = variableAccessSyntax.Name.IdentifierName;
 
+            if (variableAccessSyntax is ExplicitVariableAccessSyntax)
```

Comment:
>ExplicitVariableAccessSyntax [](http://example.com/codeflow?start=40&length=28)

I think we're forced into this approach because we don't have an intermediate representation between Bicep semantic model and the JSON.

I think we'll have to revisit this after we merge Anthony's IR PR.